src/mixins/eraser_brush.mixin.js

```diff
@@ -124,7 +124,12 @@
   fabric.util.object.extend(fabric.Object.prototype, {
     /**
      * Indicates whether this object can be erased by {@link fabric.EraserBrush}
-     * @type boolean
+     * The `deep` option introduces fine grained control over a group's `erasable` property.
+     * When set to `deep` the eraser will erase nested objects if they are erasable, leaving the group and the other objects untouched.
+     * When set to `true` the eraser will erase the entire group. Once the group changes the eraser is propagated to its children for proper functionality.
+     * When set to `false` the eraser will leave all objects including the group untouched.
+     * @tutorial {@link http://fabricjs.com/erasing#erasable_property}
+     * @type boolean | 'deep'
```

Comment:
boolean or string isn't my favorite.
With people also asking for typescript we want to keep types as much as standard as possible.
How harder it is to have this with its own parameter?

Also why the property can't stack normally?
How does it work if you have nested groups?
Seems that this deep is more a `childrenOnly` or a `deferToChildren`?

Reply:
`childrenOnly` sounds fitting.
Passing another option sounds bad to me. Makes something that should be simple error prone.
What about `enum` or `erasable | children-only | none`?

Reply:
>Also why the property can't stack normally?
>How does it work if you have nested groups?

What do mean by stack normally?
It traverses through all nested groups if they are `erasable=deep`, see [code](https://github.com/fabricjs/fabric.js/pull/7100/files#diff-d9e0aff690fbd7c86841642ab3ba18ce2cb07c9b4e87ca83c5e334f25572b9b0R659)

Reply:
BTW Typescript doesn't complain in such a case (`boolean` | `string`) . `enum` it standard in TS.

Reply:
What's your opinion?

Reply:
@asturur ?